lib/date.js

```diff
@@ -1,11 +1,63 @@
 const { views } = require('govuk-prototype-kit')
 const { DateTime, Duration, Settings } = require('luxon')
 
+const { plural } = require('./number.js')
 const { normalize } = require('./utils.js')
 
 Settings.defaultLocale = 'en-GB'
 Settings.throwOnInvalid = true
 
+/**
+ * Returns a person’s age with reference to the NHS.UK content guide
+ * Returns weeks until 6 months old.
+ * Returns months until 2 years old.
+ * Returns years above 2 years old.
+ *
+ * @see {@link https://service-manual.nhs.uk/content/inclusive-content/age}
+ * @param {string} string - Date
+ * @returns {string} Age in weeks, months or years
+ */
+function age(string) {
+  string = normalize(string, '')
+
+  try {
+    const date = new Date(string)
+
+    if (Number.isNaN(Date.parse(string))) {
+      throw new Error('Invalid DateTime')
+    }
+
+    const today = new Date()
+    const ageMs = today.valueOf() - date.valueOf()
+
+    // Return age in weeks, if less than 6 months old
+    const ageDays = ageMs / (1000 * 60 * 60 * 24)
+    if (ageDays > 0 && ageDays < 183) {
+      let ageWeeks = Math.floor(ageDays / 7)
+
+      // If less than a week old, return 1 week
+      ageWeeks = ageWeeks === 0 ? 1 : ageWeeks
+
+      return plural(ageWeeks, 'week')
+    }
+
+    // Return age in months, if less than 2 years old
+    const ageMonths =
+      (today.getFullYear() - date.getFullYear()) * 12 +
+      (today.getMonth() - date.getMonth())
+    if (ageMonths < 24) {
+      return `${ageMonths} months`
+    }
+
+    // Return age if over 2 years old
+    const ageYears = Math.floor(ageMs / 3.15576e10)
```

Comment:
The `3.15576e10` number is quite hard to follow.

Would something like this be easier to follow? Possibly better handles leap year birthday too (assuming that if you’re born on 29th Feb then your birthday isn’t until 1 March, which seems to be the UK rule?) 

```suggestion
    // Return age if over 2 years old
    const isBeforeBirthday =
      today.getMonth() < date.getMonth() ||
      (today.getMonth() === date.getMonth() && today.getDate() < date.getDate())

    const ageYears =
      today.getFullYear() - date.getFullYear() - (isBeforeBirthday ? 1 : 0)
```

Reply:
I had something like that originally, but I think – think – dividing by `3.15576e10` returns the same result? (`3.15576e10` is the average number of milliseconds in a year… a very long number).